### ISSUE TITLE
Exception creating xBim file from ifc file

### ISSUE THREAD

**Issue by brayner1** [xBimTeam/XbimEssentials#194]
#### Expected behavior:
I want to make a simple solution that opens a .ifc file, generate it's geometry, and save it as an .xbim file.

#### Actual behavior or exception details:
Actually i am receiving the following exception, on the call to model.saveAs:

> Microsoft.Isam.Esent.Interop.EsentFileAccessDeniedException: Cannot access file, the file is locked or in use
>    em Microsoft.Isam.Esent.Interop.Api.Check(Int32 err) na c:\codeplex\EsentInterop\Api.cs:linha 2937
>    em Microsoft.Isam.Esent.Interop.Api.JetCreateDatabase(JET_SESID sesid, String database, String connect, JET_DBID& dbid, CreateDatabaseGrbit grbit) na c:\codeplex\EsentInterop\Api.cs:linha 343
>    em Xbim.IO.Esent.PersistedEntityInstanceCache.CreateDatabase(String fileName) na C:\BuildAgent\work\77d7b8e93613846d\Xbim.IO\Esent\PersistedEntityInstanceCache.cs:linha 124
>    em Xbim.IO.Esent.PersistedEntityInstanceCache.ImportModel(IModel fromModel, String xbimDbName, ReportProgressDelegate progressHandler) na C:\BuildAgent\work\77d7b8e93613846d\Xbim.IO\Esent\PersistedEntityInstanceCache.cs:linha 599
>    em Xbim.IO.Esent.EsentModel.CreateFrom(IModel model, String fileName, ReportProgressDelegate progDelegate) na C:\BuildAgent\work\77d7b8e93613846d\Xbim.IO\Esent\EsentModel.cs:linha 1325
>    em Xbim.Ifc.IfcStore.SaveAs(String actualFileName, IfcStorageType actualFormat, ReportProgressDelegate progDelegate) na C:\BuildAgent\work\77d7b8e93613846d\Xbim.Ifc\IfcStore.cs:linha 795
>    em Xbim.Ifc.IfcStore.SaveAs(String fileName, Nullable`1 format, ReportProgressDelegate progDelegate) na C:\BuildAgent\work\77d7b8e93613846d\Xbim.Ifc\IfcStore.cs:linha 782
>    em ifc_to_wexBim_Converter.Program.Main(String[] args) na D:\Users\Brayner\UFPE\Voxar\Libs\ifc_to_wexBim_Converter\ifc_to_wexBim_Converter\Program.cs:linha 30

#### Versions of xBIM Packages:
xBim.Essentials: 4.0.7
xBim.Geometry: 4.0.11
log4net: 2.0.5

#### Minimal code (or steps) to reproduce the issue:
```cs
static void Main(string[] args)
        {
            const string fileName = "teklateste1.ifc";
            Console.WriteLine("Opening file");
            using (var model = IfcStore.Open(fileName))
            {
                Console.WriteLine("File opened. Creating contexts");
                var context = new Xbim3DModelContext(model);
                context.CreateContext();
                Console.WriteLine("Context created");

                var xBimFilename = Path.ChangeExtension(fileName, "xBIM");
                using (var xBiMfile = File.Create(xBimFilename))
                {
                    //using (var wexBimBinaryWriter = new BinaryWriter(xBiMfile))
                    //{
                        model.SaveAs(xBimFilename, format: Xbim.IO.IfcStorageType.Xbim);
                        //model.SaveAsWexBim(wexBimBinaryWriter);
                        //wexBimBinaryWriter.Close();
                    //}
                    xBiMfile.Close();
                }
            }
            Console.WriteLine("Function has ended");
        }
```

So, probably i'm making some mistakes to create the xbim file, since i am starting to work with xBim now. I would be grateful if someone can help me with this issue. 
Basically i'm trying to convert from .ifc to .xbim to open it in Unity, and convert the geometries for visualization in Unity. But first i need to convert the formats successfully.

**Comment by martin1cerny** [comment]
Can you try the latest develop branch? It is pretty stable now and will be published to master soon. I tried your code and it passes OK with my testing model. If it fails, can you provide us with your sample model?

**Comment by brayner1** [comment]
@martin1cerny thank you for the reply. I'l try it as soon as possible and give you the feedback. But i have one question before that: did you use the latest develop branch on the xBim.Geometry module as well? Just to make sure i repeat your steps.

**Comment by martin1cerny** [comment]
Yes I did.

**Comment by brayner1** [comment]
I'm sorry for the delay, but using the develop branch works, and i could import the model into Unity by opening the converted .xBim (with some issues, like the normals being strange in some triangles, but i'm pretty sure that's on me).
Thanks for the help.

Obs.: If someone is working with xBim in Unity, importing the .xBim models into Unity (since it is not possible to import .ifc files due to the xBim.Geometry incompatibility with Unity), i would be glad to talk about it and possibly get or deliver some help.
